src/paths/shortest_paths.c

```diff
@@ -1018,20 +1024,43 @@ int igraph_diameter(const igraph_t *graph, igraph_real_t *pres,
     if (pto != 0) {
         *pto = (igraph_integer_t) to;
     }
-    if (path != 0) {
+    if ((path_vertex) || (path_edge)) {
         if (! igraph_finite(res)) {
-            igraph_vector_clear(path);
+            if (path_vertex) {
+                igraph_vector_clear(path_vertex);
+            }
+            if (path_edge){
+                igraph_vector_clear(path_edge);
+            }
         } else {
```

Comment:
Here too, let us update the `else` branch to use `igraph_get_shortest_path` instead of `igraph_get_shortest_paths`.